controller_manager:
  ros__parameters:
    update_rate: 1000  # Hz
    
    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

arm_controller:
  ros__parameters:
    type: joint_trajectory_controller/JointTrajectoryController
    joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - end_effector_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    constraints:
      #stopped_velocity_tolerance: 0.01
      joint1:
        goal: 0.01        # Lower tolerance for better accuracy
      joint2:
          goal: 0.01
      joint3:
        goal: 0.01
      joint4:
        goal: 0.01
      end_effector_joint:
        goal: 0.01
